src/webgpu/api/operation/shader_module/compilation_info.spec.ts

```diff
@@ -0,0 +1,156 @@
+export const description = `
+ShaderModule CompilationInfo tests.
+`;
+
+import { makeTestGroup } from '../../../../common/framework/test_group.js';
+import { GPUTest } from '../../../gpu_test.js';
+import { assert } from '../../../../common/framework/util/util.js';
+
+export const g = makeTestGroup(GPUTest);
+
+const BAD_SHADER_SOURCE = `
+  [[stage(vertex)]] fn main() -> [[builtin(position)]] vec4<f32> {
+    // Expected Error: vec4 should be vec4<f32>
+    return vec4(0.0, 0.0, 0.0, 1.0);
+  }
+`;
+
+g.test('compilationInfo_valid_shader')

```

Comment:
nit: File is already called `compilation_info`, so probably no need to include `compilationInfo` in the test name (even though it's more specifically referring to `.compilationInfo()`).